### ISSUE TITLE
Code coverage audit

### ISSUE THREAD

**Issue by cwgoes** [anoma/namada#1497]
A lot of our coverage is provided by two large E2E tests (MASP & ledger transactions), which should be mostly replaced with / supplanted by unit / prop tests with proper mocks so that we can reach sufficient coverage.

**Comment by tzemanovic** [comment]
to my knowledge `make test-unit-coverage` skips e2e

**Comment by cwgoes** [comment]
Hmm - in that case @batconjurer or @mariari can you clarify the concern here?

**Comment by tzemanovic** [comment]
I agree that MASP should have unit / integration / prop tests. There's nothing to catch issues in it besides e2e tests and the lack of these tests is being a serious issue for the more complex integrations

**Comment by Fraccaman** [comment]
I think coverage is still low especially for MASP.

**Comment by tzemanovic** [comment]
I went through coverage report from #3704: https://app.codecov.io/gh/anoma/namada/pull/3704/tree/crates

In our crates, the least coverage comes from sdk (14466 missed lines), node (6593 lines), ibc (2186), apps_lib (2011). I think we can improve the coverage a lot in these if we include integration tests, which should be possible.

Specific section that need attention:
- [ ] account
	- auth & storage mods
	- priority: medium
- [ ] ibc
	- lib - a lot of the ibc functionality is only tested in e2e
	- priority: medium
- [ ] node
	- shell/governance.rs - should also be able to move this into governance crate now
	- priority: medium
- [ ] sdk
	- masp - can we also move it (with shielded-sync sub-mod) into `shielded_token`?
	- priority: high
- [ ] shielded_token
	- vp & validation mods
	- priority: high
- [ ] token
	- `Transfer` methods
	- priority: high
- [ ] tx
	- a lot of functionality inside the `types` mod is not covered 
	- it also looks like the `types` mod contains too much uncategorized logic that should be refactored
	- priority: high
